### PR TITLE
Srclink

### PR DESCRIPTION
**Pull request by TerribleDev**
* Enable debugging in Visual Studio as though you had source files locally using Source Link.
* Standardize line endings to lf to prevent debugging issues
* Ship pdb symbols in the nuget packages

Source link is a technology added to roslyn, along with a project to auto generate the files for roslyn. In short source link is a json file with urls to the source files of a project. That file is passed into roslyn as part of compilation.

The source link command line project auto generates these files based on the shaid of the current git commit, and dynamically builds links to github/bitbucket/gitlab/etc. 

In visual studio 2017, with sourcelink enabled, and debugging 'just my code' disabled. You should be able to step into these functions during interactive debugging.

srclink

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+*.cs eol=lf
```
Comment (TerribleDev) [inline .gitattributes]: we have to normalize the line endings for this to work. So if you are not publishing artefacts from a ci system, you should probably reclone the repo so your line endings are consistent.

Comment (TerribleDev) [reply .gitattributes]: well...re-clone once this is merged

**Comment by TerribleDev** [comment]
I know this was a year old but don't you think source linking is worth exploring?

**Comment by TerribleDev** [comment]
dotnet core uses the portable PDB format which is [supported](https://github.com/dotnet/sourcelink#prerequisites-for-net-projects) 

Have you tried debugging a public package? You don't just step in and have it work, you have to download the source code to disk, its infuriating to do it over and over again.

**Comment by bin3377** [comment]
okay... it's better to have feature and the framework itself is in preview. so we probably not so aggressive to absorb it now.

**Pull request closed without merging**